### PR TITLE
Fixing cors origin wildcard.

### LINKED ISSUE #5185: [Bug]: CORS issues with 3rd party web frontends when setting cors origin to * in config.json
### Requirements

- [X] Is this a bug report? For questions or discussions use https://lemmy.ml/c/lemmy_support
- [X] Did you check to see if this issue already exists?
- [X] Is this only a single bug? Do not put multiple bugs in one issue.
- [X] Do you agree to follow the rules in our [Code of Conduct](https://join-lemmy.org/docs/code_of_conduct.html)?
- [X] Is this a backend issue? Use the [lemmy-ui](https://github.com/LemmyNet/lemmy-ui) repo for UI / frontend issues.

### Summary

I noticed that there was a change in the default settings of the config.hjson and added it to our config:

https://github.com/LemmyNet/lemmy/blob/05c3eae016ffb4b96d7d2468482a93e704986d91/config/defaults.hjson#L125

However adding this makes the login in 3rd party web-frontends fail with CORS issues, kinda the opposite of what one would expect.

### Steps to Reproduce

1. Add `cors_origin: "*"` to lemmy config.hjson
2. Restart backend
3. Try to log into 3rd party web-frontends fails

### Technical Details

Running the official containers in a custom Podman setup.

### Version

BE 0.19.6

### Lemmy Instance URL

https://slrpnk.net/

### PR DESCRIPTION
**Pull request by dessalines**
- Fixes #5185

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -339,23 +339,27 @@ fn create_http_server(
...
       // Need to call send_wildcard() explicitly, passing this into allowed_origin() results in
       // error
-      if cors_origin_setting.as_deref() == Some("*") {
-        Cors::default().allow_any_origin().send_wildcard()
+      if origin == "*" {
+        cors_default
       } else {
         Cors::default()
           .allowed_origin(&origin)
           .allowed_origin(&self_origin)
```
Comment (dessalines) [inline src/lib.rs:360]: I have no idea if the `allow_any_method`, `allow_any_header` etc should also be added to this custom one.

Comment (Nothing4You) [reply src/lib.rs:360]: I wonder if this is even working without the extra settings.

> `Cors::default()`: A restrictive (security paranoid) set of defaults.
> 
> No allowed origins, methods, request headers or exposed headers. Credentials not supported. No max age (will use browser’s default).

I haven't tested setting this configuration yet, but it reads like all methods should be allowed at the very least.

I'm not sure about the headers, there are a number of headers exposed by browsers by default, but looking at requests in my browser i can see some additional headers being requested in cors preflight requests like authorization or content-type. any might be fine for this as well, sensitive headers like `set-cookie` on the response are protected anyway.

**Comment by dessalines** [comment]
Could you open that up as a separate issue and copy paste your text there? This is just for fixing the wildcard.

File: src/lib.rs (hunk 1 above)
Comment (dessalines) [reply src/lib.rs:360]: In that case I'll add the other `allows`, except for origin, to the custom origin block.

**Review by Nutomic — APPROVED** [review]

**Pull request merged**